Codes/SlippiReplays/PlaybackMenu.cpp

```diff
@@ -1,5 +1,224 @@
+        RESTORE_REGS
+        lis r12, 0x800c
+        ori r12, r12, 0x403c
+        mtctr r12
+        bctrl
+    runNormally:
+        RESTORE_REGS
+        lis r12, 0x8003
+        ori r12, r12, 0x98c8
+        mtctr r12
+        bctrl
+    )");
+    extern "C" bool processStartReplays(gfCollectionIO* collection)
+    {
+        if(startReplays.empty())
+        {
+            PopulateStartReplays();
+            collection->dataType = 7;
+            collection->dataSize = startReplays.size();
+            collection->summaryErrorCode = 0;
+            collection->fileSystemType = 0;
+            auto sizeOfNames = 20 * startReplays.size() + startReplays.size();
+            char names[sizeOfNames];
+            int index = 0;
+            for(int i = 0; i < startReplays.size(); i++)
+            {
+                for(int f = 0; f < startReplays[i].nameSize; f++)
+                {
+                    names[index] = startReplays[i].nameBuffer[f];
+                    index++;
+                }
+                names[index] = '\0';
+                index++;
+            }
+            collection->data = new char[sizeOfNames];
+            memcpy(collection->data, names, sizeOfNames);
+            OSReport("%s\n", collection->data);
+            return true;
+        }
+        return false;
+    }
+
+    INJECTION("requestSummary", 0x800c4830, R"(
+        SAVE_REGS
+        cmpwi r4, 2
+        bne runRequestSummaryNormally
+        RESTORE_REGS
+        lis r12, 0x800c
+        ori r12, r12, 0x4834
+        mtctr r12
+        bctrl
+    runRequestSummaryNormally:
+        RESTORE_REGS
+        lis r12, 0x8003
+        ori r12, r12, 0x9d40
+        mtctr r12
+        bctrl
+    )");
+
+    INJECTION("forceDispPreviewCoreReplays", 0x8119e080, "li r27, 1");
+    INJECTION("updateDispPreviewCoreReplaysLoopCount", 0x81198a50, R"(
+        mr r3, r14
+        bl getNumCharacters
+        cmplw r29, r3
+    )");
+    extern "C" u8 getNumCharacters(u8 index)
+    {
+        return startReplays[index].numPlayers;
+    }
+
+    INJECTION("updateCorrectIcons", 0x811989f8, R"(
+        mr r3, r29
+        mr r4, r14
+        bl getCurPlayerSlotID
+    )");
+
+    extern "C" u8 getCurPlayerSlotID(int playerNumber, u8 index)
+    {
+        return startReplays[index].players[playerNumber].slotID;
+    }
+
+    INJECTION("updateCorrectStageThumbnail1", 0x81198be0, R"(
+        mr r3, r14
+        bl getCurStageKind
+    )");
+
+    INJECTION("updateCorrectStageThumbnail2", 0x81198784, R"(
+        mr r3, r17
+        bl getCurStageKind
+    )");
+
+    extern "C" unsigned int getCurStageKind(u8 index)
+    {
+        return startReplays[index].stage;
+    }
+
+    INJECTION("forceDispThumbnail", 0x8119dd20, "li r0, 1");
+
+    INJECTION("printName", 0x8119e3a8, R"(
+        SAVE_REGS
+        mr r4, r14
+        bl printName
+        RESTORE_REGS
+        lwz	r12, 0x003C (r25)
+    )");
+
+    extern "C" void printName(char* title, u32 index)
```

Comment:
I like that this function makes the name nice and formatted, but I'm not convinced we should assume that the replay file follows this specific naming convention.

for example a user can simply change the file name on disk, then the substr calls could start throwing exceptions.

Maybe for now we should just show the file name as is (truncated to 20 characters though), and in a subsequent branch it could be nice to send the file modification time along as well for display in another place on the menu?